### PR TITLE
Food factory implementation

### PR DESCRIPTION
**Pull request by rpeikes**
I implemented the newInstance method

### CODE REVIEW

**Review by rpeikes — COMMENTED** [review]
Oops-the method is just called getInstance

File: src/main/java/touro/snake/FoodFactory.java [hunk 1]
```diff
@@ -10,9 +10,15 @@
...
      */
+    private final int GARDEN_WIDTH = 100;
+    private final int GARDEN_HEIGHT = 40;
+
     public Food newInstance() {
         // Peikes
-        throw new UnsupportedOperationException("Not Implemented Yet.");
+        Random coordinateGenerator = new Random();
+        int randX = coordinateGenerator.nextInt(GARDEN_WIDTH)+1;
+        int randY = coordinateGenerator.nextInt(GARDEN_HEIGHT)+1;
```
Comment (ester-agishtein) [inline src/main/java/touro/snake/FoodFactory.java:20]: Why are adding one?

Comment (rpeikes) [reply src/main/java/touro/snake/FoodFactory.java:20]: Because the Random class start generating at 0, not 1

Comment (dnagelCS) [reply src/main/java/touro/snake/FoodFactory.java:20]: I had to use your method in mine so I had to actually add it for my test to work.

Comment (rpeikes) [reply src/main/java/touro/snake/FoodFactory.java:20]: Add what?

Comment (dnagelCS) [reply src/main/java/touro/snake/FoodFactory.java:20]: I had added the method you created but I realized I probably need to mock your method instead.

**Comment by andrewoid** [comment]
@rpeikes it's called "newInstance", and you should be able to edit your comments

File: src/main/java/touro/snake/FoodFactory.java [hunk 2]
```diff
@@ -10,9 +10,15 @@
     /**
      * @return a new Food with random coordinates in the Garden
      */
+    private final int GARDEN_WIDTH = 100;
+    private final int GARDEN_HEIGHT = 40;
+
     public Food newInstance() {
         // Peikes
-        throw new UnsupportedOperationException("Not Implemented Yet.");
+        Random coordinateGenerator = new Random();
```
Comment (andrewoid) [inline src/main/java/touro/snake/FoodFactory.java:18]: This should be a constant.

File: src/main/java/touro/snake/FoodFactory.java [hunk 3]
```diff
@@ -10,9 +10,15 @@
     /**
      * @return a new Food with random coordinates in the Garden
      */
+    private final int GARDEN_WIDTH = 100;
```
Comment (andrewoid) [inline src/main/java/touro/snake/FoodFactory.java:13]: Use Garden.WIDTH instead

File: src/main/java/touro/snake/SnakeHeadStateMachine.java [hunk 4]
```diff
@@ -7,7 +7,7 @@
  * State machine that controls the turning the direction of the Snake head.
  * https://en.wikipedia.org/wiki/State_pattern
  */
-public class SnakeHeadStateMachine {
+public class  SnakeHeadStateMachine {
```
Comment (andrewoid) [inline src/main/java/touro/snake/SnakeHeadStateMachine.java:10]: remove extra space

**Pull request merged**